pkg/azureupdate/common.go

```diff
@@ -0,0 +1,86 @@
+package azureupdate
+
+import (
+	"github.com/blang/semver"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func availableReleases(g8sclient versioned.Interface) ([]*semver.Version, error) {
+	releaseList, err := g8sclient.ReleaseV1alpha1().Releases().List(v1.ListOptions{})
+	if err != nil {
+		return []*semver.Version{}, microerror.Mask(err)
+	}
+
+	var ret []*semver.Version
+	for _, release := range releaseList.Items {
+		parsed, err := semver.ParseTolerant(release.Name)
+		if err != nil {
+			return []*semver.Version{}, microerror.Maskf(invalidReleaseError, "Unable to parse release %s to a semver.Release", release.Name)
+		}
+		ret = append(ret, &parsed)
+	}
+
+	return ret, nil
+}
+
+func clusterIsUpgrading(cr *v1alpha1.AzureConfig) (bool, string) {
+	for _, cond := range cr.Status.Cluster.Conditions {
+		if cond.Type == conditionUpdating {
+			return true, conditionUpdating
+		}
+		if cond.Type == conditionCreating {
+			return true, conditionCreating
+		}
+	}
+
+	return false, ""
+}
+
+func included(releases []*semver.Version, release *semver.Version) bool {
+	for _, r := range releases {
+		if r.EQ(*release) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func upgradeAllowed(g8sclient versioned.Interface, oldVersion *semver.Version, newVersion *semver.Version) (bool, error) {
+	if !oldVersion.Equals(*newVersion) {
```

Comment:
this function allows _upgrading_ to the same version. Is that what we want?

Reply:
If you don't allow that you are not able to edit the Cars ever again after you create them